airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.scheduler.persistence;
+
+import com.google.common.collect.Lists;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.JobCheckConnectionConfig;
+import io.airbyte.config.JobConfig;
+import io.airbyte.config.JobDiscoverCatalogConfig;
+import io.airbyte.config.JobGetSpecConfig;
+import io.airbyte.config.JobOutput;
+import io.airbyte.config.JobSyncConfig;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardSyncOutput;
+import io.airbyte.config.State;
+import io.airbyte.scheduler.Attempt;
+import io.airbyte.scheduler.AttemptStatus;
+import io.airbyte.scheduler.Job;
+import io.airbyte.scheduler.ScopeHelper;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class DefaultJobCreator implements JobCreator {
+
+  private JobPersistence jobPersistence;
+
+  public DefaultJobCreator(JobPersistence jobPersistence) {
+    this.jobPersistence = jobPersistence;
+  }
+
+  @Override
+  public long createSourceCheckConnectionJob(SourceConnection source, String dockerImageName) throws IOException {
+    final String scope =
+        ScopeHelper.createScope(
```

Comment:
What is required already to get rid of this scope concept?

Reply:
the goal of the scope is to be able to group jobs that are created from the same config. so really any other solution that does that works fine.

Reply:
i haven't through of one yet that doesn't really closely resemble what we already have or that is much better. admittedly haven't spent too much time on this yet. this hasn't been a pain point at all. you have a specific looming concern?